src/amo/utils/index.js

```diff
@@ -48,3 +50,34 @@ export const getCanonicalURL = ({
 |}): string => {
   return `${_config.get('baseURL')}${locationPathname}`;
 };
+
+type QueryParams = { [key: string]: any };
+
+type MakeInternalExternalURLParams = {|
+  _config: typeof config,
+  urlString: string,
+  internalQueryParams?: QueryParams,
+  externalQueryParams?: QueryParams,
+|};
+
+export const makeInternalExternalURL = ({
+  _config = config,
+  externalQueryParams = {},
+  internalQueryParams = {},
```

Comment:
maybe you should hard-code the default internal and external params and change the name of the function to something like `addParamsToPromoUrl()`?